File: frontend/src/interfaces/OrderItem.ts
import type { Order } from "./Order";
import type { GameKey } from "./GameKey";

export interface OrderItem {
  ID: number;
  unit_price: number;
  qty: number;
  line_discount: number;
  line_total: number;

  order_id: number;
  order?: Order;

  game_key_id?: number | null;
  game_key?: GameKey | null;
}

export interface CreateOrderItemRequest {
  unit_price: number;
  qty: number;
  line_discount: number;
  order_id: number;
  game_key_id?: number | null;
}

export interface UpdateOrderItemRequest {
  ID: number;
  unit_price?: number;
  qty?: number;
  line_discount?: number;
  line_total?: number;
  game_key_id?: number | null;
}

File: backend/entity/order_item.go
// entity/order_item.go
package entity

import "gorm.io/gorm"

type OrderItem struct {
	gorm.Model
	UnitPrice   float64 `json:"unit_price"`
	QTY         int     `json:"qty"`
	LineDiscount float64 `json:"line_discount"`
	LineTotal   float64 `json:"line_total"`

	OrderID uint   `json:"order_id"`
	Order   *Order `gorm:"foreignKey:OrderID" json:"order,omitempty"`

	// ผูกคีย์เกมที่เบิกให้รายการนี้ (ถ้ามี)
	GameKeyID *uint    `json:"game_key_id"`
	GameKey   *GameKey `gorm:"foreignKey:GameKeyID" json:"game_key,omitempty"`
}
